xbmc/filesystem/CurlFile.cpp

```diff
@@ -748,7 +748,8 @@ void CCurlFile::ParseAndCorrectUrl(CURL &url2)
 
   // lookup host in DNS cache
   std::string resolvedHost;
-  if (CServiceBroker::GetDNSNameCache()->GetCached(url2.GetHostName(), resolvedHost))
+  auto dnsCache = CServiceBroker::GetDNSNameCache();
```

Comment:
Replace `auto`by actual type `std::shared_ptr<CDNSNameCache>`?
Maybe this even could be `const std::shared_ptr<CDNSNameCache>`?
Why the new nullptr check? Did you experience any crashes? Isn't the cache supposed to be always available?